### PR TITLE
Redo PR #7430 for new website (docs for WAN federation over mesh gateways)

### LINKED ISSUE #7416: document wan federation via mesh gateways
This is the followup issue for the documentation changes for https://github.com/hashicorp/consul/issues/6884

- [ ] describe interaction with https://github.com/hashicorp/consul/issues/7159
- [ ] website updates
- [ ] learn guide?
- [ ] upgrade guide from 1.7.x: (note: should call out that you need to first complete the upgrade of all consul servers everywhere and the clients co-located with mesh gateways as a precondition before toggling the newfangled federation on to avoid just such a problem)
- [ ] ensure agent options page lists correct consul version for the feature (`This was added in Consul 1.8.x **TODO(wanfed)**`)
- [ ] document that secondaries need `operator:write` on their replication token

### PR DESCRIPTION
**Pull request by preetapan**
Fixes #7416 

Diagrams will come in a future PR

### CODE REVIEW

File: website/pages/docs/connect/wan_federation_via_mesh_gateways.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+# WAN Federation via Mesh Gateways <sup>Beta</sup>
+
+-> **1.8.0+:**  This feature is available in Consul versions 1.8.0 and higher
+
+~> This topic requires familiarity with [mesh gateways](/docs/connect/mesh_gateway).
+
+WAN federation via mesh gateways allows for Consul servers in different datacenters
+to be federated exclusively through mesh gateways.
+
+When setting up a
```
Comment (lkysow) [inline website/pages/docs/connect/wan_federation_via_mesh_gateways.mdx:18]: Should we add a link (if you're using Kubernetes, see /docs/k8s/installation/multi-cluster)

File: website/pages/docs/connect/wan_federation_via_mesh_gateways.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
+---
```
Comment (lkysow) [inline website/pages/docs/connect/wan_federation_via_mesh_gateways.mdx:1]: I don't think this file should be underscore separated, I think it should be dash separated  to match the rest  of the docs

File: website/pages/docs/connect/wan-federation-via-mesh-gateways.mdx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+WAN federation via mesh gateways allows for Consul servers in different datacenters
+to be federated exclusively through mesh gateways.
+
+When setting up a
+[multi-datacenter](https://learn.hashicorp.com/consul/security-networking/datacenters)
+Consul cluster, operators must ensure that all Consul servers in every
+datacenter must be directly connectable over their WAN-advertised network
+address from each other.
+
+If you are using Kubernetes, refer to our [Kubernetes Multi Cluster](/docs/k8s/installation/multi-cluster) documentation.
```
Comment (preetapan) [inline website/pages/docs/connect/wan-federation-via-mesh-gateways.mdx:24]: @lkysow this is more of a placeholder line for now, feels like the wrong place to say this here, seems like it should be in its own section or maybe the first sentence of this topic. thoughts?

Comment (lkysow) [reply website/pages/docs/connect/wan-federation-via-mesh-gateways.mdx:24]: I think this is fine here but could also live in an alert at the top of the page.

**Pull request merged**